osu.Game.Rulesets.Osu/Mods/OsuModHidden.cs

```diff
@@ -7,24 +7,33 @@
 using osu.Framework.Graphics;
 using osu.Game.Rulesets.Mods;
 using osu.Game.Rulesets.Objects.Drawables;
-using osu.Game.Rulesets.Objects.Types;
 using osu.Game.Rulesets.Osu.Objects;
 using osu.Game.Rulesets.Osu.Objects.Drawables;
+using osu.Game.Rulesets.Objects.Types;
+using osu.Game.Configuration;
+using osu.Framework.Configuration;
 
 namespace osu.Game.Rulesets.Osu.Mods
 {
-    public class OsuModHidden : ModHidden, IApplicableToDrawableHitObjects
+    public class OsuModHidden : ModHidden, IApplicableToDrawableHitObjects, IReadFromConfig
     {
         public override string Description => @"Play with no approach circles and fading notes for a slight score advantage.";
         public override double ScoreMultiplier => 1.06;
-
         private const double fade_in_duration_multiplier = 0.4;
         private const double fade_out_duration_multiplier = 0.3;
+        private Bindable<bool> increaseFirstObjectVisibility = new Bindable<bool>();
+
+        public void ReadFromConfig(OsuConfigManager config)
+        {
+            increaseFirstObjectVisibility = config.GetBindable<bool>(OsuSetting.IncreaseFirstObjectVisibility);
+        }
 
         public void ApplyToDrawableHitObjects(IEnumerable<DrawableHitObject> drawables)
         {
             foreach (var d in drawables.OfType<DrawableOsuHitObject>())
             {
+                //Don't hide the first object if the increaseFirstObjectVisibility is true ("drawables" are in a reverse order -> Last() )
+                if (d == drawables.Last() && increaseFirstObjectVisibility) continue;

```

Comment:
We should probably be moving this to the base implementation. I.e.:

```
class ModHidden
{
    public void ApplyToDrawableHitObjects(IEnumerable<DrawableHitObject> drawables)
    {
        foreach (var d in drawables.OfType<DrawableHitObject>())
        {
             if (d == drawables.Last() && increaseHitObjectVisibility)
                 return;
             ApplyHiddenState(d)
        }
    }

    protected virtual void ApplyHiddenState(DrawableHitObject hitObject) { }
}

Reply:
The hidden mods for the other gamemodes do not need to hide (or not) the first hidden object, only the standard gamemode need it, so should we really implement it in a class (ModHidden) that all the hidden mods inherit?

Reply:
The idea was to bring all rulesets in-line with this behaviour, so @smoogipoo's approach would be correct here.